miden-lib/asm/miden/kernels/tx/tx.masm

```diff
@@ -276,38 +279,38 @@ export.create_note
     # => [tag, note_type, RECIPIENT]
 
     # get the index for the next note to be created and increment counter
-    exec.increment_num_created_notes
-    # => [note_idx, tag, note_type, RECIPIENT]
+    exec.increment_num_created_notes dup movdn.7
+    # => [note_idx, tag, note_type, RECIPIENT, note_idx]
 
     # get a pointer to the memory address at which the note will be stored
     exec.memory::get_created_note_ptr
-    # => [note_ptr, tag, note_type, RECIPIENT]
+    # => [note_ptr, tag, note_type, RECIPIENT, note_idx]
 
     # populate the metadata
     swap exec.account::get_id movup.3
-    # => [note_type, sender_acct_id, tag, note_ptr, RECIPIENT]
+    # => [note_type, sender_acct_id, tag, note_ptr, RECIPIENT, note_idx]
 
     push.0 # TODO: allow the user to push the aux data
```

Comment:
Let's create an issue to address this TODO.

Reply:
https://github.com/0xPolygonMiden/miden-base/issues/731